src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -38,6 +39,9 @@
 public class SamReaderFactoryTest extends HtsjdkTest {
```

Comment:
We should also have a test here somehow for the fallback from URI -> URL or whatever we ultimately decide to do.